crates/control/src/calibration_controller.rs

```diff
@@ -0,0 +1,299 @@
+                    *context.field_dimensions,
+                );
+
+                info!("Calibration complete! Corrections: {solved_result:?}");
+                self.last_calibration_corrections = Some(solved_result);
+                Some(CalibrationCommand::FINISH)
+            }
+            CalibrationCommand::FINISH => None,
+        }
+    }
+
+    fn try_transition_to_look_at(&mut self, dispatch_time: CycleTime) -> CalibrationCommand {
+        let next_look_at = self.get_next_look_at();
+
+        match next_look_at {
+            Some((target, camera)) => CalibrationCommand::LOOKAT {
+                camera,
+                target,
+                dispatch_time,
+            },
+            None => {
+                info!("\tNothing else to LOOKAT, goto PROCESS");
+                CalibrationCommand::PROCESS
+            }
+        }
+    }
+
+    fn get_next_look_at(&mut self) -> Option<(Point2<Ground>, Option<CameraPosition>)> {
+        let current_index = self.current_look_at_index;
+        self.current_look_at_index += 1;
+        if current_index < self.look_at_list.len() {
+            Some(self.look_at_list[current_index])
+        } else {
+            None
+        }
+    }
+}
+
+fn collect_filtered_values(
+    measurement_perception_input: &PerceptionInput<Vec<&CalibrationCaptureResponse<Measurement>>>,
+    original_dispatch_time: &CycleTime,
+) -> (Option<Measurement>, bool) {
+    let result = measurement_perception_input
+        .persistent
+        .iter()
+        .flat_map(|(_cycle_timestamp, measurements)| measurements.iter())
+        .find(|&measurement| match measurement {
+            CalibrationCaptureResponse::CommandRecieved {
+                dispatch_time,
+                output: _,
+            } => original_dispatch_time.start_time == dispatch_time.start_time,
+            CalibrationCaptureResponse::Idling => false,
+            CalibrationCaptureResponse::RetriesExceeded { dispatch_time } => {
+                original_dispatch_time.start_time == dispatch_time.start_time
+            }
+        });
+
+    let mut retries_exceeded = false;
+    let mut value_found = false;
+    let measurement: Option<Measurement> = result.and_then(|value| match value {
+        CalibrationCaptureResponse::Idling => None,
+        CalibrationCaptureResponse::CommandRecieved {
+            dispatch_time: _,
+            output,
+        } => {
+            if output.is_some() {
+                value_found = true;
+                output.clone()
+            } else {
+                None
+            }
+        }
+        CalibrationCaptureResponse::RetriesExceeded { dispatch_time: _ } => {
+            retries_exceeded = true;
+            None
+        }
+    });
+
+    (measurement, retries_exceeded || value_found)
+}
+
+// TODO Add fancier logic to either set this via parameters OR detect the location, walk, etc
+fn generate_look_at_list() -> Result<Vec<(Point2<Ground>, Option<CameraPosition>)>> {
```

Comment:
Why is the camera position optional?

Reply:
Current look-at behavious has it optional, for convenience, I left it optional in the context of look-at command.

Since the system works in this state, I'll make it mandatory and tighten the control.

Reply:
Done